app/models/deleted_resource.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+##
+# Model to store deleted resources.
+# These may be used in order to serve '410 Gone' statuses.
+class DeletedResource
+  include Mongoid::Document
+
+  belongs_to :deleted_by, class_name: 'User', optional: true, inverse_of: :deleted_resources,
+                          index: true
+
+  field :resource_type, type: String
+  field :resource_uuid, type: String
```

Comment:
Could we also have a field `record_id` to store the record's MongoDB ID?

(To support also storing Contributions as DeletedResources without a UUID.)